zkevm-circuits/src/evm_circuit/table.rs

```diff
@@ -0,0 +1,284 @@
+                [
+                    tag,
+                    F::from(value),
+                    F::from((value >> 7) * 0xFFu64),
+                    F::zero(),
+                ]
+            })),
+            Self::BitwiseAnd => Box::new((0..256).flat_map(move |lhs| {
+                (0..256).map(move |rhs| {
+                    [tag, F::from(lhs), F::from(rhs), F::from(lhs & rhs)]
+                })
+            })),
+            Self::BitwiseOr => Box::new((0..256).flat_map(move |lhs| {
+                (0..256).map(move |rhs| {
+                    [tag, F::from(lhs), F::from(rhs), F::from(lhs | rhs)]
+                })
+            })),
+            Self::BitwiseXor => Box::new((0..256).flat_map(move |lhs| {
+                (0..256).map(move |rhs| {
+                    [tag, F::from(lhs), F::from(rhs), F::from(lhs ^ rhs)]
+                })
+            })),
+            Self::ResponsibleOpcode => Box::new(
+                ExecutionState::iterator().flat_map(move |execution_state| {
+                    execution_state.responsible_opcodes().into_iter().map(
+                        move |opcode| {
+                            [
+                                tag,
+                                F::from(execution_state.as_u64()),
+                                F::from(opcode.as_u64()),
+                                F::zero(),
+                            ]
+                        },
+                    )
+                }),
+            ),
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum TxContextFieldTag {
+    Nonce = 1,
+    Gas,
+    GasTipCap,
+    GasFeeCap,
+    CallerAddress,
+    CalleeAddress,
+    IsCreate,
+    Value,
+    CallDataLength,
+    CallData,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum RwTableTag {
+    TxAccessListAccount = 1,
+    TxAccessListStorageSlot,
+    TxRefund,
+    Account,
+    AccountStorage,
+    AccountDestructed,
+    CallContext,
+    Stack,
+    Memory,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum AccountFieldTag {
+    Nonce = 1,
+    Balance,
+    CodeHash,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum CallContextFieldTag {
+    RwCounterEndOfReversion = 1,
+    CallerCallId,
+    TxId,
+    Depth,
+    CallerAddress,
+    CalleeAddress,
+    CallDataOffset,
+    CallDataLength,
+    ReturnDataOffset,
+    ReturnDataLength,
+    Value,
+    Result,
+    IsPersistent,
+    IsStatic,
+
+    IsRoot,
+    IsCreate,
+    OpcodeSource,
+    ProgramCounter,
+    StackPointer,
+    GasLeft,
+    MemorySize,
+    StateWriteCounter,
+}
+
+impl_expr!(FixedTableTag);
+impl_expr!(TxContextFieldTag);
+impl_expr!(RwTableTag);
+impl_expr!(AccountFieldTag);
+impl_expr!(CallContextFieldTag);
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub(crate) enum Table {
+    Fixed,
+    Tx,
+    Rw,
+    Bytecode,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum Lookup<F> {
+    Fixed {
+        tag: Expression<F>,
```

Comment:
Add some doc to each field in these lookup table?

Reply:
Added in [`3c57d83`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/3c57d83839d68064d81fea52ef9a55ec82a3ef5b).